src/forms/TextInput.jsx

```diff
@@ -5,18 +5,14 @@ import cx from 'classnames';
 /**
  * @module TextInput
  */
-class TextInput extends React.Component {
+class TextInput extends React.PureComponent {
 
 	constructor(props) {
 		super(props);
-		this.state = {
-			value: props.value || '',
-		};
 		this.onChange = this.onChange.bind(this);
 	}
 
 	onChange(e) {
-		this.setState({ value: e.target.value });
 		if (this.props.onChange) {
 			this.props.onChange(e);
```

Comment:
hm ok. Well what do we prefer? `PureComponent` vs stateless functional component? What are the benefits?
@eilinora  should we go for the latter instead of a PureComponent?